lib/token.cpp

```diff
@@ -2388,24 +2385,24 @@ const ValueFlow::Value* Token::getMaxValue(bool condition, MathLib::bigint path)
 
 const ValueFlow::Value* Token::getMovedValue() const
 {
-    if (!mImpl->mValues)
+    if (values().empty())
         return nullptr;
-    const auto it = std::find_if(mImpl->mValues->begin(), mImpl->mValues->end(), [](const ValueFlow::Value& value) {
+    const auto it = std::find_if(values().begin(), values().end(), [](const ValueFlow::Value& value) {
         return value.isMovedValue() && !value.isImpossible() &&
         value.moveKind != ValueFlow::Value::MoveKind::NonMovedVariable;
     });
-    return it == mImpl->mValues->end() ? nullptr : &*it;
+    return it == values().end() ? nullptr : &*it;
 }
 
 // cppcheck-suppress unusedFunction
 const ValueFlow::Value* Token::getContainerSizeValue(const MathLib::bigint val) const
 {
-    if (!mImpl->mValues)
+    if (values().empty())
```

Comment:
`values().empty()` is redundant here.

Reply:
I kept it to avoid the function calls (which obviously won't do much) since some of these might be called millions of times. I haven't profiled any of this yet.

Reply:
I would not call it "redundant" but "unnecessary". Still needs to be profiled.